### ISSUE TITLE
Not responding to adds behind the character

### ISSUE THREAD

**Issue by thejefu** [Xian55/WowClassicGrindBot#570]
### Description

When botting, I'm occasionally taking aggro from adds which are placed behind my character.

When this happens, the character pauses indefinitely as if waiting to leave combat.

In the logs attached, this happened at the end of the log. I manually targeted the add and resolved the conflict before ending the bot.

### Addon Version

Unknown

### World of Warcraft Client

The Burning Crusade Classic

### Reproduction Steps

Bot until you experience an aggro from behind.

### Logs
[out2024_0215_001.log](https://github.com/user-attachments/files/15518471/out2024_0215_001.log)

**Comment by Xian55** [comment]
Did the character were idle for 20 seconds ?

```
[11:19:17:961 I] [Shoot ] instant input 77ms CAST_START -79.4555ms
[11:19:18:057 I] [Homunculi ] ... BeforeCastDelay 2000ms
[11:19:18:537 I] [GoapAgent ] Kill credit detected! Known kills: 1 | Fighting with: 1
[11:19:20:198 I] [Homunculi ] instant input 63ms NONE -64.3067ms
[11:19:20:325 I] [Homunculi ] instant input 62ms NONE -64.1518ms
[11:19:20:341 W] [GoapAgent ] New Plan= NO PLAN
[11:19:45:111 I] [BagChangeTracker] + 1 Malachite
[11:19:46:161 I] [CombatGoal ] Turning too fast!
[11:19:46:177 I] [Power Word: Shield] ... BeforeCastDelay 2000ms
[11:19:48:392 I] [Power Word: Shield] instant usable: True->False | current: False->True | NONE->NONE
```
I will do some tests today on the PTR.

```
[11:19:20:341 W] [GoapAgent ] New Plan= NO PLAN
```
No plan means that for the given situation there was no suitable Goal to execute, during that time the player character either idle.

**Comment by thejefu** [comment]
Correct, the malachite + turning too fast was my looting followed by the correction.

**Comment by Xian55** [comment]
Today did some extensive testing, i had an issue during my character was using Wand, for some reason the player just wasn't faced to the target.

Not sure how to fully tackle the issue as i have yet to found a 100% repro case. :(

**Comment by Applesafter** [comment]
Hi,
Sorry to Necro this issue slightly, I have been testing the bot on the Anniversary Realms and it seems to work really well for the most part so thank you for your effort. This issue "not responding to adds" is the only consistent issue I have. If the bot pulls another mob while in combat it will quite frequently get stuck targeting the intial mobs corpse and get bashed to death by the second mob. All the time the dashboard will be on the "Combat" stage with all conditions in green.

If you set "Skin" to "True" it will quite often spam skin the initial mob while getting constantly interupted by the second mob.

I'm not sure if this is something you can fix, I can try to more provide logs if you need them as the repro rate, for me at least, is reasonably high in denser areas. The issue seems to happen around line 113.
[Death to add log.txt](https://github.com/user-attachments/files/17907112/Death.to.add.log.txt)

**Comment by Xian55** [comment]
Thanks for the feedback i will look into it, in the upcoming days!

**Comment by Applesafter** [comment]
Downloaded the new version of the bot and tested, still happening.  Thanks for trying.
[log7.txt](https://github.com/user-attachments/files/18010902/log7.txt)

**Comment by Xian55** [comment]
Potential fix was supplied with #639

**Comment by Xian55** [comment]
Closing due there are other reports meanwhile #651 please continue the discussion there.
